### PR TITLE
Task-3.1

### PR DESCRIPTION
**Pull request by eduardo-neto-epam**
- Added postgresql elephantSql DB;
- Added Sequelize ORM with migrations, seed files and config with sequilizerc;
- Added scripts to run sequelize cli commands;
- Updated code to work with new database.

### CODE REVIEW

**Comment by eduardo-neto-epam** [comment]
@Dmitry-White 
Task 3.2 requests a patten that I already follow although I have more of a components approach. 
Do I need to change anything here or is it done already?

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -39,57 +34,74 @@ class UserController implements IController {
...
         }
     };
 
     deleteUser = async (request: Request, response: Response, next: NextFunction): Promise<void> => {
         const { id } = request.params;
         try {
-            const deletedUser = await this.usersDb.delete(id);
-            deletedUser ? response.send(deletedUser.id) : next(new UserNotFoundException(id));
+            const userId = await this.userService.deleteUser(id);
+            if (userId instanceof HttpException) {
```
Comment (Dmitry-White) [inline src/user/user.controller.ts:88]: Consider using "early-return" approach here.

File: src/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                throw new HttpException(StatusCodes.INTERNAL_SERVER_ERROR, error.message);
+            }
+            throw new HttpException(StatusCodes.INTERNAL_SERVER_ERROR, 'Oops, Something went wrong');
+        }
+    };
+
+    findById = async (id: string): Promise<User | UserNotFoundException> => {
+        try {
+            const user = await this.userModel.findByPk(id);
+            return user === null || user.getDataValue('isDeleted') ? new UserNotFoundException(id) : user;
```
Comment (Dmitry-White) [inline src/user/user.service.ts:31]: Consider using "early-return" approach here.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]
Looks pretty solid, seems like you've got some experience already!

File: src/database/config/database.config.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { Sequelize } from 'sequelize';
+
+const db = new Sequelize(process.env.DB_URL as string, {
```
[the remark this answers is not in the document]
Comment (eduardo-neto-epam) [reply src/database/config/database.config.ts:3]: Thanks! :)

**Review by Dmitry-White — APPROVED** [review]
Looks great!

**Pull request merged**